modules/electrophysiology_browser/ajax/get_electrophysiology_session_data.php

```diff
@@ -0,0 +1,336 @@
+                LEFT JOIN session s ON (s.ID=pf.SessionID) 
+                LEFT JOIN candidate c USING (CandID)
+                LEFT JOIN psc ON (s.CenterID=psc.CenterID) 
+                LEFT JOIN physiological_output_type pot 
+                  USING (PhysiologicalOutputTypeID) 
+              WHERE 
+                s.Active = "Y" 
+                AND pf.FileType IN ("bdf", "cnt", "edf", "set", "vhdr", "vsm") 
+              ORDER BY pf.SessionID';
+
+    $response = array();
+
+    $sessions            = $db->pselect($query, array());
+    $sessions            = array_column($sessions, 'SessionID');
+    $response['patient'] = getSubjectData($sessionID);
+    $response['database']    = array_values(getFilesData($sessionID));
+    $response['sessions']    = $sessions;
+    $currentIndex            = array_search($sessionID, $sessions);
+    $response['nextSession'] = $sessions[$currentIndex+1] ?? '';
+    $response['prevSession'] = $sessions[$currentIndex-1] ?? '';
+
+    return $response;
+}
+
+/**
+ * Get the subject data information.
+ *
+ * @param int $sessionID ID of the electrophysiology session
+ *
+ * @return array with the subject information
+ */
+function getSubjectData(int $sessionID)
+{
+    $subjectData = array();
+    $timePoint   = \NDB_Factory::singleton()->timepoint($sessionID);
+    $candidate   = \NDB_Factory::singleton()->candidate($timePoint->getCandID());
+
+    $subjectData['pscid']       = $candidate->getPSCID();
+    $subjectData['dccid']       = $candidate->getCandID();
+    $subjectData['visit_label'] = $timePoint->getVisitLabel();
+    $subjectData['sessionID']   = $sessionID;
+    $subjectData['site']        = $timePoint->getPSC();
+    $subjectData['dob']         = $candidate->getCandidateDoB();
+    $subjectData['sex']         = $candidate->getCandidateSex();
+    $subjectData['subproject']  = $timePoint->getData('SubprojectTitle');
+    $subjectData['output_type'] = htmlentities(
+        $_REQUEST['outputType'],
+        ENT_QUOTES,
+        "UTF-8"
+    );
+
+    return $subjectData;
+}
+
+/**
+ * Get the list of electrophysiology recordings with their recording information.
+ *
+ * @param int $sessionID ID of the electrophysiology session
+ *
+ * @return array with the file collection
+ */
+function getFilesData(int $sessionID)
+{
+    $db = \NDB_Factory::singleton()->database();
+
+    $fileCollection = array();
+    $params         = array();
+    $outputType     = $_REQUEST['outputType'];
+    $params['SID']  = $sessionID;
+    $query          = 'SELECT 
+                         pf.PhysiologicalFileID, 
+                         pf.FilePath 
+                       FROM 
+                         physiological_file pf ';
+
+    if ($outputType != 'all_types') {
+        $query .= 'LEFT JOIN physiological_output_type pot ON ';
+        $query .= 'pf.PhysiologicalOutputTypeID=pot.PhysiologicalOutputTypeID ';
+        $query .= 'WHERE SessionID=:SID ';
+        if ($outputType == 'raw') {
```

Comment:
if the output_type is validated at the start of the file, this can be streamline to `$query .= 'AND pot.OutputTypeName = "' . $outputType . '"';`

Reply:
done!